### PR TITLE
Fix Bloomberg extractor. (closes #10630)

### LINKED ISSUE #10630: [Bloomberg] Broken site.
- [X] I've **verified** and **I assure** that I'm running youtube-dl **2016.09.11.1**
### Before submitting an _issue_ make sure you have:
- [X] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
### What is the purpose of your _issue_?
- [X] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---
### If the purpose of this _issue_ is a _bug report_, _site support request_ or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
$ youtube-dl -v https://www.bloomberg.com/gadfly/articles/2016-09-01/tesla-what-big-oil-has-to-fear
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'https://www.bloomberg.com/gadfly/articles/2016-09-01/tesla-what-big-oil-has-to-fear']
[debug] Encodings: locale UTF-8, fs UTF-8, out None, pref UTF-8
[debug] youtube-dl version 2016.09.11.1
[debug] Python version 2.7.6 - Linux-4.2.0-42-generic-x86_64-with-Ubuntu-14.04-trusty
[debug] exe versions: avconv 9.18-6, avprobe 9.18-6, rtmpdump 2.4
[debug] Proxy map: {}
[Bloomberg] tesla-what-big-oil-has-to-fear: Downloading webpage
ERROR: tesla-what-big-oil-has-to-fear: Failed to parse JSON  (caused by ValueError('Expecting property name enclosed in double quotes: line 2 column 11 (char 12)',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/home/abuchbinder/Downloads/youtube-dl/youtube_dl/extractor/common.py", line 549, in _parse_json
    return json.loads(json_string)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    obj, end = self.scan_once(s, idx)
ValueError: Expecting property name enclosed in double quotes: line 2 column 11 (char 12)
Traceback (most recent call last):
  File "/home/abuchbinder/Downloads/youtube-dl/youtube_dl/YoutubeDL.py", line 691, in extract_info
    ie_result = ie.extract(url)
  File "/home/abuchbinder/Downloads/youtube-dl/youtube_dl/extractor/common.py", line 347, in extract
    return self._real_extract(url)
  File "/home/abuchbinder/Downloads/youtube-dl/youtube_dl/extractor/bloomberg.py", line 52, in _real_extract
    r'BPlayer\(null,\s*({[^;]+})\);', webpage, 'id'), name)
  File "/home/abuchbinder/Downloads/youtube-dl/youtube_dl/extractor/common.py", line 553, in _parse_json
    raise ExtractorError(errmsg, cause=ve)
ExtractorError: tesla-what-big-oil-has-to-fear: Failed to parse JSON  (caused by ValueError('Expecting property name enclosed in double quotes: line 2 column 11 (char 12)',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

---
### Description of your _issue_, suggested solution and other information

It looks like the problem is that `json.loads` only accepts double-quotes, but the source from Bloomberg looks like this:

```
...
        BPlayer(null, {
          'id': '303188263',
          'idType': 'AVMM',
          'htmlChildId': 'player-303188263',
          'autoplay': true,
...
```

Adding a `.replace("'", '"')` to the string would probably work, but I don't know what the local style is here. I'll see if I can figure out how to propose a patch.

### PR DESCRIPTION
**Pull request by adambuchbinder**
## Please follow the guide below
- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your _pull request_ (like that [x])
- Use _Preview_ tab to see how your _pull request_ will actually look like

---
### Before submitting a _pull request_ make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
### What is the purpose of your _pull request_?
- [X] Bug fix
- [ ] New extractor
- [ ] New feature

---
### Description of your _pull request_ and other information

Explanation of your _pull request_ in arbitrary form goes here. Please make sure the description explains the purpose and effect of your _pull request_ and is worded well enough to be understood. Provide as much context and examples as possible.

The JSON being sent from Bloomberg uses single quotes, which is invalid
JSON, and contains function calls. Regex both of those out.

Additionally, the API endpoint requires an additional parameter, and
must be called over HTTPS now. Change that as well.

Signed-off-by: Adam Buchbinder adam.buchbinder@gmail.com

### CODE REVIEW

File: youtube_dl/extractor/bloomberg.py [hunk 1]
```diff
@@ -48,13 +48,20 @@ def _real_extract(self, url):
             r'["\']bmmrId["\']\s*:\s*(["\'])(?P<url>.+?)\1',
             webpage, 'id', group='url', default=None)
         if not video_id:
-            bplayer_data = self._parse_json(self._search_regex(
-                r'BPlayer\(null,\s*({[^;]+})\);', webpage, 'id'), name)
+            bplayer_json = self._search_regex(r'BPlayer\(null,\s*({[^;]+})\);',
+                webpage, 'id')
+            # It's not good JSON; it uses single quotes and contains function
```
Comment (TRox1972) [inline youtube_dl/extractor/bloomberg.py]: Are you sure it isn't Javascript code instead of JSON? If so, you can use `js_to_json`

Comment (adambuchbinder) [inline youtube_dl/extractor/bloomberg.py]: Unfortunately, that doesn't work. I get `"bb_region": "getCookie"("ak_rg")` in the "JSON".

**Comment by adambuchbinder** [comment]
Test URLs:

http://www.bloomberg.com/politics/videos/2014-11-27/surviving-by-imbibing
https://www.bloomberg.com/gadfly/articles/2016-09-01/tesla-what-big-oil-has-to-fear

File: youtube_dl/extractor/bloomberg.py [hunk 2]
```diff
@@ -47,14 +47,23 @@ def _real_extract(self, url):
...
-            bplayer_data = self._parse_json(self._search_regex(
-                r'BPlayer\(null,\s*({[^;]+})\);', webpage, 'id'), name)
+        if video_id:
+            video_id_type = 'BMMR'
+        else:
+            bplayer_json = self._search_regex(r'BPlayer\(null,\s*({[^;]+})\);',
+                webpage, 'id')
+            # It's not good JSON; it uses single quotes and contains function
+            # calls. Sweep that under the rug.
+            bplayer_json = bplayer_json.replace("\'", '"')
```
Comment (yan12125) [inline youtube_dl/extractor/bloomberg.py]: Use ```js_to_json``` instead.

**Comment by yan12125** [comment]
Could you add a test case to ```_TESTS```? Also squash commits into a single one when everything is done.

**Pull request closed without merging**